### PR TITLE
Allows to use Maturin with PDM

### PR DESCRIPTION
**Pull request by Tpt**

### CODE REVIEW

File: kraken-std/src/kraken/std/python/buildsystem/maturin.py [hunk 1]
```diff
@@ -196,14 +148,89 @@ def build(self, output_directory: Path, as_version: str | None = None) -> list[P
...
             pyproject.set_version(old_poetry_version)
             pyproject.raw.save()
 
         return dst_files
 
 
-class MaturinManagedEnvironment(PoetryManagedEnvironment):
+class MaturinPoetryPyprojectHandler(PoetryPyprojectHandler):
+    def set_version(self, version: str | None) -> None:
+        PyprojectHandler.set_version(self, version)
```
Comment (NiklasRosenstein) [inline kraken-std/src/kraken/std/python/buildsystem/maturin.py:145]: Shouldn't be necessary as it's called by the Poetry implementation is necessary?

File: kraken-std/src/kraken/std/python/buildsystem/maturin.py [hunk 2]
```diff
@@ -212,3 +239,56 @@ def install(self, settings: PythonSettings) -> None:
 
     def always_install(self) -> bool:
         return True
+
+
+class MaturinPdmPythonBuildSystem(PDMPythonBuildSystem):
+    """A maturin-backed version of the PDM build system, that invokes the maturin build-backend.
+    Can be enabled by adding the following to the local pyproject.yaml:
+    ```toml
+    ~[tool.pdm.dev-dependencies]
```
Comment (NiklasRosenstein) [inline kraken-std/src/kraken/std/python/buildsystem/maturin.py]: Tilde intentional?

File: kraken-std/src/kraken/std/python/buildsystem/maturin.py [hunk 3]
```diff
@@ -78,86 +52,51 @@ class MaturinZigTarget:
...
-        handler = self.get_pyproject_reader(pyproject)
-        handler.synchronize_project_section_to_poetry_state()
-
-    def build(self, output_directory: Path, as_version: str | None = None) -> list[Path]:
+    def build(
+        self,
+        output_directory: Path,
+        as_version: str | None,
+        get_pyproject_reader: Callable[[Pyproject], PyprojectHandler],
+        project_directory: Path,
```
Comment (ntrinquier) [inline kraken-std/src/kraken/std/python/buildsystem/maturin.py]: I know they are only used in `build`, but would it make sense to store `get_pyproject_reader` and `project_directory` (potentially `output_directory` too) in the constructor to simplify `build`'s signature?

File: kraken-std/src/kraken/std/python/buildsystem/maturin.py (hunk 1 above)
Comment (Tpt) [reply kraken-std/src/kraken/std/python/buildsystem/maturin.py:145]: Sadly the Poetry implementation is not calling `super()`. So, it is not changing the version in the `[project]` section but only in the `[tool.poetry]` section.

File: kraken-std/src/kraken/std/python/buildsystem/maturin.py (hunk 2 above)
Comment (Tpt) [reply kraken-std/src/kraken/std/python/buildsystem/maturin.py]: Indeed not. Fixed. Thank you!

**Review by ntrinquier — APPROVED** [review]

**Pull request merged**